### DISCUSSION TITLE
Update Report for checking for lonely cables

### DISCUSSION THREAD

**Discussion by SteveHauke** [netbox-community/netbox#6826] [Q&A] [answered]
Hello,

I had a Netbox report, which showed me all cables, who are not connected to an endpoint (device).
Unfortunately  the function **cable.get_path_endpoints()** is no longer available.
Can someone help me, to get this running again?

Here is my old code:
`from dcim.constants import *
from dcim.models import Cable
from extras.reports import Report

class Cables(Report):
    description = "Check if cables make sense by being connected to endpoints"
    def test_cables(self):
        for cable in Cable.objects.all():
            if str(cable.termination_a_type) == 'dcim | front port':
                if str(cable.termination_b_type) == 'dcim | front port':
                    if str(cable.status) == 'connected':
                        if 'None' in str(cable.get_path_endpoints()): 
                            self.log_failure(cable, "Cable is not connected to two endpoints")
                        else:
                            self.log_success(cable)
`

**Comment by candlerb** [comment, marked as answer]
You want to check whether the path which the cable is part of, ends on an *Interface* at both ends, rather than on a frontport or rearport.

Look at the code [here](https://github.com/netbox-community/netbox/blob/v2.11.9/netbox/dcim/api/views.py#L79-L90) which identifies all the paths which pass through the termination (in general, if it were a rearport rather than a frontport, multiple paths could pass through it).

I believe the CablePath will only exist if there are interfaces at both ends.  The condition then becomes something like this (you may need to experiment a bit with it in nbshell):

```
if CablePath.objects.filter(path__contains=cable.termination_a).count() == 0:
    self.log_failure(cable, "Cable is not connected to two endpoints")
```

I'd also point out that looking at string representations of objects is a fragile way to check their type.  You could do something like this instead:

```
if isinstance(c.termination_a, FrontPort) and isinstance(c.termination_b, FrontPort):
    ...etc
```

**Reply by SteveHauke** [reply]
Thank you very much.
This is my working script now and I hope that there is every cable in it.

`from dcim.constants import *
from dcim.models import *
from extras.reports import Report

class Cables(Report):
	description = "Check if cables make sense by being connected to endpoints"
	def test_cables(self):

		for cable in Cable.objects.all():
			if CablePath.objects.filter(path__contains=cable.termination_a).count() == 0:
				if CablePath.objects.filter(path__contains=cable.termination_b).count() == 0:
					if isinstance(cable.termination_a, FrontPort) or isinstance(cable.termination_b, FrontPort):
						self.log_failure(cable, "Cable is not connected to two endpoints")`

**Reply by candlerb** [reply]
Excellent.  It may run a little faster if you move

```
if isinstance(cable.termination_a, FrontPort) or isinstance(cable.termination_b, FrontPort):
```

to be the first condition in the loop, since this can be done without a database query - so it will quickly skip cables that go from interface to interface, console cables, power cables etc.
